### PR TITLE
Add getStatusName

### PR DESCRIPTION
**Pull request by TAOXUY**
**Description:**
Add util function to convert enum `Status` to string name.

**Why:**
It can be used as error name in the respond code detail in the envoy [jwt_authn filter](https://github.com/envoyproxy/envoy/tree/master/source/extensions/filters/http/jwt_authn)

### CODE REVIEW

**Comment by TAOXUY** [comment]
@qiwzhang @nareddyt PTAL

File: jwt_verify_lib/status.h [hunk 1]
```diff
@@ -219,6 +219,13 @@ enum class Status {
  */
 std::string getStatusString(Status status);
 
+/**
+ * Convert enum status to string name.
+ * @param status is the enum status.
+ * @return the status name.
+ */
+std::string getStatusName(Status status);
```
Comment (qiwzhang) [inline jwt_verify_lib/status.h:227]: Why is this needed?  Could you describe use case?

File: src/status.cc [hunk 2]
```diff
@@ -176,5 +176,161 @@ std::string getStatusString(Status status) {
   return "";
 }
 
+
+std::string getStatusName(Status status) {
```
Comment (nareddyt) [inline src/status.cc:180]: Probably better to fill in the response code details in the upstream filter.

File: jwt_verify_lib/status.h (hunk 1 above)
Comment (TAOXUY) [reply jwt_verify_lib/status.h:227]: Follow the convention of response code details which is one concise word in snake case.

File: src/status.cc (hunk 2 above)
Comment (TAOXUY) [reply src/status.cc:180]: I was kind of debating.

Prefer to put this util function in this lib as it looks like a common usage like `getStatusString`.

File: jwt_verify_lib/status.h (hunk 1 above)
Comment (qiwzhang) [reply jwt_verify_lib/status.h:227]: Try to use getStatusString().  If not enough, we can improve it.

Comment (qiwzhang) [reply jwt_verify_lib/status.h:227]: The enum name doesn't mean much.  Its string should carry detail info.  Can we just put the StatusString into rc_detail.

Comment (TAOXUY) [reply jwt_verify_lib/status.h:227]: Agree it isn't as informative but almost all the filters use this kind of condensed word as reponse_code_detail.

I think it is doable . Lemme have a PR and see if upstream envoy accept it.

**Comment by TAOXUY** [comment]
Use the statusString for rc_detail instead. This PR is no need anymore.
